src/main/java/net/countercraft/movecraft/combat/movecraftcombat/listener/IgniteListener.java

```diff
@@ -0,0 +1,86 @@
+package net.countercraft.movecraft.combat.movecraftcombat.listener;
+
+
+import com.sk89q.worldedit.bukkit.BukkitWorld;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldguard.WorldGuard;
+import com.sk89q.worldguard.protection.ApplicableRegionSet;
+import com.sk89q.worldguard.protection.flags.Flags;
+import com.sk89q.worldguard.protection.flags.StateFlag;
+import com.sk89q.worldguard.protection.managers.RegionManager;
+import com.sk89q.worldguard.protection.regions.ProtectedRegion;
+import net.countercraft.movecraft.Movecraft;
+import net.countercraft.movecraft.combat.movecraftcombat.config.Config;
+import net.countercraft.movecraft.config.Settings;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.CraftManager;
+import net.countercraft.movecraft.utils.MathUtils;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.WorldGuard6Utils;
+import net.countercraft.movecraft.combat.movecraftcombat.utils.LegacyUtils;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockIgniteEvent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class IgniteListener implements Listener {
+    @EventHandler(priority = EventPriority.LOW)
+    public void onBlockIgnite(BlockIgniteEvent event) {
+        if (event.isCancelled()) {
+            return;
+        }
+        final Craft adjacentCraft = adjacentCraft(event.getBlock().getLocation());
```

Comment:
If AddFiresToHitbox and FireballPenetration are disabled, this is a costly function to run on every block ignition.

Instead I would recommend calling a doFireballPenetration function if it is enabled and the cause is a fireball, and after that, I would call doAddFiresToHitbox if it is enabled.

Infact, fireball penetration does not use the adjacent Craft, so it's calculation can be done in doAddFiresToHitbox.